cli/bpmetadata/repo.go

```diff
@@ -24,23 +24,31 @@ const (
 
 // getRepoDetailsByPath takes a local path for a blueprint and tries
 // to get repo details that include its name, path and type
-func getRepoDetailsByPath(bpPath string, sourceUrl *BlueprintRepoDetail) (*repoDetail, error) {
+func getRepoDetailsByPath(bpPath string, sourceUrl *BlueprintRepoDetail, repoName string, readmeContent []byte) *repoDetail {
 	rootRepoPath := getBpRootPath(bpPath)
 	if sourceUrl == nil {
 		bpPath = strings.TrimSuffix(bpPath, "/")
 		repoUrl, err := util.GetRepoUrl(bpPath)
```

Comment:
Looks like GetRepoUrl returns repoUrl as "" if there is an error, so can probably drop the entire `err != nil` block?

```suggestion
		repoUrl, _ := util.GetRepoUrl(bpPath)
```

Reply:
Agreed but then the linter will complain. The right thing to do is not to have it returns errors at all since we're bypassing that now but that requires a bit more testing. I'm going to keep this for now and do another PR to take that out after testing all possible regressions.

Reply:
SG!